File: c++/basic_grammar/operation/assignment_operator/subtraction.cpp
#include <iostream>
using namespace std;

int subtraction_assignment_operator() {
  int a = 0;
  for (int i = 0; i < 5; i++) {
    a -= 2;
  }
  return a;
}

int back_subtraction_assignment_operator() {
  int a = 0;
  for (int i = 0; i < 5; i++) {
    a--;
  }
  return a;
}

int front_subtraction_assignment_operator() {
  int a = 0;
  for (int i = 0; i < 5; i++) {
    --a;
  }
  return a;
}

double decimal_subtraction_assignment_operator() {
  double a = 0;
  for (int i = 0; i < 5; i++) {
    a -= 1.5;
  }
  return a;
}

int negative_subtraction_assignment_operator() {
  int a = 0;
  for (int i = 0; i < 5; i++) {
    a -= -2;
  }
  return a;
}

int main() {
  cout << subtraction_assignment_operator() << endl;
  cout << back_subtraction_assignment_operator() << endl;
  cout << front_subtraction_assignment_operator() << endl;
  cout << decimal_subtraction_assignment_operator() << endl;
  cout << negative_subtraction_assignment_operator() << endl;
  return 0;
}

File: typescript/basic_grammar/operation/assignment_operator/subtraction.ts
export const subtractionAssignmentOperator = () => {
  let a = 0
  for (let i = 0; i <  5; i++) {
    a -= 2
  }
  return a
}

export const backSubtractionAssignmentOperator = () => {
  let a = 0
  for (let i = 0; i < 5; i++) {
    a--
  }
  return a
}

export const frontSubtractionAssignmentOperator = () => {
  let a = 0
  for (let i = 0; i < 5; i++) {
    --a
  }
  return a
}

export const decimalSubtractionAssignmentOperator = () => {
  let a = 0
  for (let i = 0; i < 5; i++) {
    a -= 1.5
  }
  return a
}

export const negativeSubtractionAssignmentOperator = () => {
  let a = 0
  for (let i = 0; i < 5; i++) {
    a -= -2
  }
  return a
}

console.log(subtractionAssignmentOperator())
console.log(backSubtractionAssignmentOperator())
console.log(frontSubtractionAssignmentOperator())
console.log(decimalSubtractionAssignmentOperator())
console.log(negativeSubtractionAssignmentOperator())
